community/modules/scripts/wait-for-startup/main.tf

```diff
@@ -13,22 +13,42 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+locals {
+  instance_names = toset(compact(concat(var.instance_names, [var.instance_name])))
+}
+
+resource "null_resource" "validate_instance_names" {
+  lifecycle {
+    precondition {
```

Comment:
I think you could put this precondition on the lifecycle of the data block and avoid having a separate null_resource

Reply:
Done.